### PR TITLE
[Docs] Application runtime clarifications

### PR DESCRIPTION
**Pull request by alonmr**

### CODE REVIEW

**Review by liranbg — APPROVED** [review]

File: docs/runtimes/application.ipynb [hunk 1]
```diff
@@ -59,17 +62,36 @@
...
+    "# Optionally add mlrun\n",
     "application.deploy(with_mlrun=False)"
    ]
+  },
+  {
+   "cell_type": "markdown",
+   "source": [
+    "Reusing an already built reverse proxy image is done when:\n",
+    "1. Redeploying a function that built the reverse proxy once and has `application.status.container_image` enriched.\n",
+    "2. After it was built manually with `mlrun.runtimes.ApplicationRuntime.deploy_reverse_proxy_image()`.\n",
```
Comment (jillnogold) [inline docs/runtimes/application.ipynb]: "- Redeploying a function that built the reverse proxy once and has `application.status.container_image` enriched.\n",
    "- It was already built manually with `mlrun.runtimes.ApplicationRuntime.deploy_reverse_proxy_image()`.\n",

File: docs/runtimes/application.ipynb [hunk 2]
```diff
@@ -12,6 +12,8 @@
     "The application runtime deploys a container image (for example, a web application) that is exposed on a specific port, and a command to run the HTTP server. The runtime is based on top of Nuclio, and adds the application as a side-car to a Nuclio function pod while the actual function is a reverse proxy to that application. \n",
     "\n",
     "You can set an existing image to run in the application, or let the application runtime build the side-car image for you, by specifying the source code, and pulling at build-time. \n",
+    "\n",
+    "> Note: The default base image is python:3.9 when not specifying and image for the application.\n",
```
Comment (jillnogold) [inline docs/runtimes/application.ipynb]: "> Note: The default base image is python 3.9 when not specifying and image for the application.\n",

Comment (alonmr) [reply docs/runtimes/application.ipynb]: `python:3.9` is the name of the image. its a docker convention. I'll mark it as code to make it clear

**Review by jillnogold — APPROVED** [review]

File: docs/runtimes/application.ipynb [hunk 3]
```diff
@@ -59,17 +62,36 @@
...
    ]
+  },
+  {
+   "cell_type": "markdown",
+   "source": [
+    "Reusing an already built reverse proxy image is done when:\n",
+    "- Redeploying a function that built the reverse proxy once and has `application.status.container_image` enriched.\n",
+    "- It was already built manually with `mlrun.runtimes.ApplicationRuntime.deploy_reverse_proxy_image()`.\n",
+    "\n",
+    "Using one of the above options can help minimize redundant builds and streamline the development cycle. "
```
Comment (TomerShor) [inline docs/runtimes/application.ipynb:90]: I'm thinking of adding a section on how to do it in air-gapped systems, but in a followup PR

**Review by TomerShor — APPROVED** [review]
LGTM

**Pull request merged**